### PR TITLE
Remove common

### PR DESCRIPTION
**Pull request by bquantump**
Remove Common/ManagementTestShared
Remove RecordedTestBase{Tenviroment} (no need for the useless wrapper I think)

Move ManagementTestShared functionally into record test base

Classes that did use ManagementTestShared now use RecordedTestBase

Did not fix import yet (to resource manager)

This is all in effort to reduce size of PR to implement #15434 in the future

### CODE REVIEW

File: sdk/core/Azure.Core.TestFramework/src/ClientTestBase.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ public abstract class ClientTestBase
         private static readonly IInterceptor s_diagnosticScopeValidatingInterceptor = new DiagnosticScopeValidatingInterceptor();
         private static Dictionary<Type, Exception> s_clientValidation = new Dictionary<Type, Exception>();
         public bool IsAsync { get; }
-
+        public RecordedTestMode Mode { get; set; }
```
Comment (weshaggard) [inline sdk/core/Azure.Core.TestFramework/src/ClientTestBase.cs:24]: Why was this necessary here and in the RecordedTestAttribute?

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs [hunk 2]
```diff
@@ -9,20 +9,27 @@
...
 
         protected RecordMatcher Matcher { get; set; }
 
         public TestRecording Recording { get; private set; }
 
-        public RecordedTestMode Mode { get; set; }
+        private static TimeSpan ZeroPollingInterval { get; } = TimeSpan.FromSeconds(0);
 
+        protected ResourceGroupCleanupPolicy CleanupPolicy { get; set; }
+
```
Comment (weshaggard) [inline sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:31]: Should this be a little more then just the clean-up policy? It would be good if it could be a factory so it can be in charge of creating, tracking, and cleaning up the resource groups.

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs [hunk 3]
```diff
@@ -154,5 +166,65 @@ public virtual void StopTestRecording()
...
+                {
+                    if (resType.ResourceType == resourceType)
+                        return true;
+                    else
+                        return false;
+                }
+                ).First().Locations.FirstOrDefault();
+        }
+
+        protected void SleepInTest(int milliSeconds)
```
Comment (weshaggard) [inline sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:223]: What cases do we want to sleep? In general we try to avoid sleeping but instead usually do some sort of polling.

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs (hunk 2 above)
Comment (bquantump) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:31]: I have not change the names of anything yet. I just wanted to get everything moved over (out of common/ManagementTestShared) first in a separate PR with minimal changes to remove ManagmentRecordTestBase, RecordedTestBase{TEnviroment} and moving ResourceGroupCleanupPolicy

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs (hunk 3 above)
Comment (bquantump) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:223]: Again just moved it over for now. Can change in separate PR that will have the larger changes discussed. This is just to reduce the size of that future PR.

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs [hunk 4]
```diff
@@ -9,20 +9,27 @@
...
+    where TEnvironment : TestEnvironment, new()
     {
         protected RecordedTestSanitizer Sanitizer { get; set; }
 
         protected RecordMatcher Matcher { get; set; }
 
         public TestRecording Recording { get; private set; }
 
-        public RecordedTestMode Mode { get; set; }
+        private static TimeSpan ZeroPollingInterval { get; } = TimeSpan.FromSeconds(0);
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:28]: This has a single call site I think we can inline the usage.

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase{TEnvironment}.cs [hunk 5]
```diff
@@ -1,31 +0,0 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
-namespace Azure.Core.TestFramework
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/RecordedTestBase{TEnvironment}.cs:4]: Why was this removed?

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs [hunk 6]
```diff
@@ -9,20 +9,27 @@
...
 using NUnit.Framework.Interfaces;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.ResourceManager.Resources;
 
 namespace Azure.Core.TestFramework
 {
     [Category("Recorded")]
-    public abstract class RecordedTestBase : ClientTestBase
+    public abstract class RecordedTestBase<TEnvironment> : ClientTestBase
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:19]: Why did we merge two RecordedTestBase classes?

File: sdk/core/Azure.Core.TestFramework/src/ResourceGroupCleanupPolicy.cs [hunk 7]
```diff
@@ -7,7 +7,7 @@
 using Azure.Core;
 using Azure.Core.Pipeline;
 
-namespace Azure.ResourceManager.TestFramework
+namespace Azure.Core.TestFramework
 {
     public class ResourceGroupCleanupPolicy : HttpPipelineSynchronousPolicy
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/ResourceGroupCleanupPolicy.cs:12]: This implementation has to be thread-safe.

Comment (pakrym) [reply sdk/core/Azure.Core.TestFramework/src/ResourceGroupCleanupPolicy.cs:12]: Access to ResourceGroupsCreated is not syncronized.

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs [hunk 8]
```diff
@@ -154,5 +166,65 @@ public virtual void StopTestRecording()
...
+            CleanupPolicy = new ResourceGroupCleanupPolicy();
+            options.AddPolicy(CleanupPolicy, HttpPipelinePosition.PerCall);
+
+            return CreateClient<ResourcesManagementClient>(
+                TestEnvironment.SubscriptionId,
+                TestEnvironment.Credential,
+                options);
+        }
+
+        protected async Task CleanupResourceGroupsAsync()
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:194]: Who calls this?

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs [hunk 9]
```diff
@@ -154,5 +166,65 @@ public virtual void StopTestRecording()
...
+                    TestEnvironment.Credential,
+                    new ResourcesManagementClientOptions()).ResourceGroups;
+                foreach (var resourceGroup in CleanupPolicy.ResourceGroupsCreated)
+                {
+                    await resourceGroupsClient.StartDeleteAsync(resourceGroup);
+                }
+            }
+        }
+
+        protected async Task<string> GetFirstUsableLocationAsync(ProvidersOperations providersClient, string resourceProviderNamespace, string resourceType)
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:209]: Shouldn't we use the location from the test environment?

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestAttribute.cs [hunk 10]
```diff
@@ -83,7 +83,7 @@ private static bool IsTestFailedWithRecordingMismatch(TestExecutionContext conte
             }
         }
 
-        private static void SetRecordMode(RecordedTestBase fixture, RecordedTestMode mode)
+        private static void SetRecordMode(ClientTestBase fixture, RecordedTestMode mode)
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/RecordedTestAttribute.cs:86]: I don't think we should have `Mode` on the `ClientTestBase `

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs (hunk 4 above)
Comment (bquantump) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:28]: Do we want to make these type of changes now? I just wanted to this stuff moved over from common really.

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase{TEnvironment}.cs (hunk 5 above)
Comment (bquantump) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase{TEnvironment}.cs:4]: I wanted to see why this was here the first place. It seems like a useless layer on top of recordedTestBase. I can put it back, but what is the reason for this extra layer here?

File: sdk/core/Azure.Core.TestFramework/src/ResourceGroupCleanupPolicy.cs (hunk 7 above)
Comment (bquantump) [reply sdk/core/Azure.Core.TestFramework/src/ResourceGroupCleanupPolicy.cs:12]: Why would this change, the namespace change, change the behavior of this class from its current behavior, or are you saying this is currently a problem with this class?

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs (hunk 9 above)
Comment (bquantump) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:209]: Just moved this method over from Common/ManagementTestShared.

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase{TEnvironment}.cs (hunk 5 above)
Comment (pakrym) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase{TEnvironment}.cs:4]: So you don't need to move Mode into the ClientTestBase

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestAttribute.cs (hunk 10 above)
Comment (bquantump) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestAttribute.cs:86]: This was so we could remove RecordedTestBase{TEviroment}.cs. If we have reason to keep that class, can remove this.

File: sdk/core/Azure.Core.TestFramework/src/ResourceGroupCleanupPolicy.cs (hunk 7 above)
Comment (pakrym) [reply sdk/core/Azure.Core.TestFramework/src/ResourceGroupCleanupPolicy.cs:12]: There is currently a problem with the class.

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs (hunk 4 above)
Comment (pakrym) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:28]: This won't be a large change and would decrease the diff size.

Comment (bquantump) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:27]: We are staying with parity with this PR

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs (hunk 2 above)
Comment (bquantump) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:30]: We are staying with parity with this PR

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs (hunk 9 above)
Comment (bquantump) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:178]: We are staying with parity with this PR

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs (hunk 3 above)
Comment (bquantump) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:192]: We are staying with parity with this PR

File: sdk/core/Azure.Core.TestFramework/src/ResourceGroupCleanupPolicy.cs (hunk 7 above)
Comment (bquantump) [reply sdk/core/Azure.Core.TestFramework/src/ResourceGroupCleanupPolicy.cs:12]: We are staying with parity with this PR

**Comment by bquantump** [comment]
@weshaggard @pakrym 

What is this coverage.cobertura.xml breaking here: https://dev.azure.com/azure-sdk/public/_build/results?buildId=638345&view=logs&j=1e00682f-6081-5348-8da2-7d601664b2e3&t=511c56c3-2035-5753-cb8d-07ad2fe3e1b8&l=25

**Comment by pakrym** [comment]
>What is this coverage.cobertura.xml breaking here: dev.azure.com/azure-sdk/public/_build/results?buildId=638345&view=logs&j=1e00682f-6081-5348-8da2-7d601664b2e3&t=511c56c3-2035-5753-cb8d-07ad2fe3e1b8&l=25

@heaths, can you take a look?

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestAttribute.cs (hunk 10 above)
Comment (pakrym) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestAttribute.cs]: Yes, being able to reference a recorded test class without knowing the generic argument is the reason to keep the `RecordedTestBase` and avoid this change.

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs [hunk 11]
```diff
@@ -140,9 +147,10 @@ public virtual void StartTestRecording()
             if (Mode != RecordedTestMode.Live &&
                 test.Properties.ContainsKey("SkipRecordings"))
             {
-                throw new IgnoreException((string) test.Properties.Get("SkipRecordings"));
+                throw new IgnoreException((string)test.Properties.Get("SkipRecordings"));
             }
             Recording = new TestRecording(Mode, GetSessionFilePath(), Sanitizer, Matcher);
+            // Set the TestEnvironment Mode here so that any Mode changes in RecordedTestBase are picked up here also.
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:153]: Is this comment relevant?

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs [hunk 12]
```diff
@@ -21,6 +24,10 @@ public abstract class RecordedTestBase : ClientTestBase
 
         public TestRecording Recording { get; private set; }
 
+        private static TimeSpan ZeroPollingInterval { get; } = TimeSpan.FromSeconds(0);
+
+        protected ResourceGroupCleanupPolicy CleanupPolicy { get; set; }
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:29]: Does this property have to be in the `RecordedTestBase`? It's only initialized in the generic version of the class?

File: sdk/core/Azure.Core.TestFramework/src/Azure.Core.TestFramework.csproj [hunk 13]
```diff
@@ -11,6 +11,7 @@
     <PackageReference Include="Microsoft.NET.Test.Sdk" />
     <PackageReference Include="Moq" />
     <Reference Include="System.Web" Condition="'$(TargetFramework)' == 'net461' OR '$(TargetFramework)' == 'net47'"/>
+     <ProjectReference Include="../../../resources/Azure.ResourceManager.Resources/src/Azure.ResourceManager.Resources.csproj" />
```
Comment (heaths) [inline sdk/core/Azure.Core.TestFramework/src/Azure.Core.TestFramework.csproj:14]: All track 2 data-plane test projects use this project. Perhaps only include this in test projects for mgmt-plane? This will complicate the dependency graph MSBuild will have to calculate and slow builds. Perhaps go the other way: have this project reference automatically include Azure.Core.TestFramework to limit scope.

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs (hunk 3 above)
Comment (heaths) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:189]: I recommend a tracking bug you can link from these comments. It's a fairly consistent way we use to keep track of improvements to make from PRs.

**Review by heaths — COMMENTED** [review]
Please open at least one issue to track some of the comments made here you said you'd fix in a subsequent PR.

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs (hunk 3 above)
Comment (bquantump) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:189]: Good idea @heaths 

I will make issues for these when we make further changes in the future.

File: sdk/core/Azure.Core.TestFramework/src/Azure.Core.TestFramework.csproj (hunk 13 above)
Comment (bquantump) [reply sdk/core/Azure.Core.TestFramework/src/Azure.Core.TestFramework.csproj:14]: I'm thinking the best way to do this may be as was suggested by either @weshaggard or @pakrym , forgot who. That is, to make a small rest client to do this for us, since we do not want cross dependencies and the unit tests that run in Azure.ResourceManager.Resources can get out of a sync with a released nuget package, which will cause these test fail making it fragile to import a nuget package here (since this test framework is depended a upon by Azure.ResourceManager.Resources).

**Comment by jsquire** [comment]
Hi @bquantump.  There hasn't been recent engagement on this PR.  Would you please be so kind as to let us know if this is still an active work stream or if the PR should be closed out?

**Comment by bquantump** [comment]
> Hi @bquantump. There hasn't been recent engagement on this PR. Would you please be so kind as to let us know if this is still an active work stream or if the PR should be closed out?

Will move to draft since the solution to the cross reference will be to generate parts of the Resource management client into the test framework

**Comment by jsquire** [comment]
Hi @bquantump.  There hasn't been recent engagement on this PR.  Would you please be so kind as to let us know if this is still an active work stream by removing the `no-recent-activity` label?  Otherwise, we'll close this out in 7 days.

**Comment by jsquire** [comment]
Since there hasn't been recent engagement, this is being closed out.  Feel free to reopen if you'd like to continue working on these changes.  Please be sure to remove the `no-recent-activity` label otherwise, this will be closed again in the next cleanup pass.

**Comment by jsquire** [comment]
Since there hasn't been recent engagement, this is being closed out.  Feel free to reopen if you'd like to continue working on these changes.  Please be sure to remove the `no-recent-activity` label otherwise, this will be closed again in the next cleanup pass.

**Comment by jsquire** [comment]
Hi @bquantump.  There hasn't been recent engagement on this PR.  If this is still an active work stream, please let us know by removing the `no-recent-activity` label.  Otherwise, we'll close this out in 7 days.
